src/parser.ts

```diff
@@ -2611,9 +2611,15 @@ export class Parser {
 
     parseCatchClause(): Node.CatchClause {
         const node = this.createNode();
+        let body;
 
         this.expectKeyword('catch');
 
+        if (!this.match('(')) {
+            body = this.parseBlock();
```

Comment:
I would rather not duplicate this part. Just set `param` to `null` and skip the parameter parsing.